asmcomp/selectgen.ml

```diff
@@ -194,18 +195,139 @@ let join_array rs =
 (* Name of function being compiled *)
 let current_function_name = ref ""
 
+(* Environment parameter for the function being compiled, if any. *)
+let current_function_env_param = ref None
```

Comment:
This could still be made less hackish by using the environment in cmmgen to do that: in Cmmgen.get_field, if ptr is the current environment variable, then generate a immutable cload.